### PR TITLE
Refactored find.match function

### LINKED ISSUE #120: Refactor `find.match.R`
- [ ] reformat
- [ ] extract functions
- [ ] rename variables
- [ ] reuse functions
- [ ] Improve documentation

### PR DESCRIPTION
**Pull request by zargham-ahmad**
This PR contains the following:

- extracted function, renamed variables, and updated some logic to make it more readable

closes #120

### CODE REVIEW

File: R/find.match.R [hunk 1]
```diff
@@ -1,38 +1,38 @@
+find_min_position <- function(distances) {
+    position <- which(distances == min(distances))[1]
```
Comment (hechth) [inline R/find.match.R]: use which.min

File: R/find.match.R [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
-            p<-find.min.pos(a)
+#'
+#' @param distances A matrix of distances.
+#' @param max_distance A distance larger than which cannot be accepted as pairs.
+#' @return A binary matrix the same size as the input matrix, with matched position taking value 1, and all other positions taking value 0.
+find.match <- function(distances, max_distance) {
+    matches <- matrix(0, nrow(distances), ncol(distances))
+
+    if (ncol(distances) == 1) {
+        sel <- which(distances[, 1] == min(distances[, 1]))[1]
```
Comment (hechth) [inline R/find.match.R]: use which.min

**Review by hechth — COMMENTED** [review]
apply suggested comments, otherwise looks good.

File: R/find.match.R (hunk 1 above)
Comment (zargham-ahmad) [reply R/find.match.R]: resolved in [4fd428f](https://github.com/RECETOX/recetox-aplcms/pull/193/commits/4fd428f79d85f15e2760d2755843dff298e9ec98)

File: R/find.match.R (hunk 2 above)
Comment (zargham-ahmad) [reply R/find.match.R]: resolved in [4fd428f](https://github.com/RECETOX/recetox-aplcms/pull/193/commits/4fd428f79d85f15e2760d2755843dff298e9ec98)

**Pull request merged**